### PR TITLE
feat: add journey visitor export to journeys-admin

### PR DESCRIPTION
**Pull request by tataihono**
# PR Description

## 🎯 Overview

This PR implements the frontend UI for journey visitor export functionality, allowing users to export journey visitor data with their associated events through an intuitive interface in the journeys-admin application.

## 📋 Summary

Implements the complete frontend interface for the journey visitor export feature, building upon the backend GraphQL query `journeyVisitorExport` that was previously implemented in [ENG-3266](https://linear.app/jesus-film-project/issue/ENG-3266/add-journey-visitor-export-with-events-inline). The implementation provides users with a comprehensive export dialog that supports both visitor actions and contact data exports.

### 🎛️ Export Dialog Interface
- **Dual Export Types**: Support for both "Visitor Actions" and "Contact Data" exports
- **Date Range Selection**: Configurable start and end dates for export filtering
- **Dynamic Form Controls**: Context-sensitive form fields based on selected export type
- **Loading States**: Proper loading indicators during export operations
- **Error Handling**: User-friendly error messages for failed exports

### 📊 Visitor Actions Export
- **Event Type Selection**: Filter by specific event types (RadioQuestionSubmissionEvent, SignUpSubmissionEvent, TextResponseSubmissionEvent)
- **Date Filtering**: Export events within specified date ranges
- **CSV Format**: Structured CSV output with proper headers and data formatting

### 👥 Contact Data Export
- **Field Selection**: Choose specific contact fields (name, email, phone)
- **Event Data Integration**: Include associated event data in the export
- **Selective Export**: Granular control over which contact and event data to include
- **Smart Filtering**: Automatic exclusion of certain fields to maintain data structure integrity

### Backend Integration
- **GraphQL Query**: Integration with `journeyVisitorExport` query
- **Schema Updates**: Added `JourneyVisitorExportSelect` input type for field selection
- **Authorization**: Proper ACL checks to ensure only authorized users can access export functionality

### Frontend Components
- **ExportDialog**: Main dialog component with export type selection and configuration
- **ContactDataForm**: Specialized form for contact data field selection with checkboxes
- **DateRangePicker**: Reusable component for date range selection
- **FilterForm**: Component for visitor action event type selection

### State Management
- **Local State**: Component-level state management for form data
- **Loading States**: Separate loading states for different export types
- **Error Handling**: Comprehensive error handling with user feedback

## 🧪 Testing

- **Unit Tests**: Comprehensive test coverage for all new components
- **Integration Tests**: Tests for export functionality and user interactions
- **Mock Implementations**: Proper mocking of GraphQL queries and hooks for testing
- **Edge Cases**: Testing for empty data, error scenarios, and boundary conditions

### Backend Changes
- `apis/api-journeys-modern/schema.graphql` - Added export query and input types
- `apis/api-journeys-modern/src/schema/journeyVisitor/` - Enhanced visitor export functionality
- `libs/prisma/journeys/db/` - Database schema updates

### Frontend Changes
- `apps/journeys-admin/src/components/JourneyVisitorsList/FilterDrawer/ExportDialog/` - Main export dialog implementation
- `apps/journeys-admin/src/libs/useJourneyContactsExport/` - Contact data export hook and utilities
- Enhanced CSV processing and download functionality

## 🎯 Acceptance Criteria Met

- ✅ Export button/option available in journey visitor management interface
- ✅ Users can configure export filters (event types, date ranges)
- ✅ CSV file downloads successfully with proper filename
- ✅ Loading states shown during export process
- ✅ Error messages displayed for failed exports
- ✅ Only authorized users can access export functionality
- ✅ UI is responsive and accessible

## 🔗 Related Issues

- **Linear Issue**: [NES-875](https://linear.app/jesus-film-project/issue/NES-875/implement-journey-visitor-export-ui-in-frontend)
- **Backend Implementation**: [ENG-3266](https://linear.app/jesus-film-project/issue/ENG-3266/add-journey-visitor-export-with-events-inline)
- **Related PR**: [PR #7922](https://github.com/JesusFilm/core/pull/7922) (Backend implementation)

## 🚀 Deployment Notes

- No breaking changes to existing functionality
- Backward compatible with existing journey visitor management
- Requires backend changes from ENG-3266 to be deployed first
- No additional environment variables or configuration required

## Summary by CodeRabbit

- New Features
  - Contact Data export with selectable fields (Name, Email, Phone, Poll Selection, Subscription, Text Submission), including Select All and sensible defaults.
- Improvements
  - CSV exports now download directly with a dated filename.
  - Visitor Actions CSV includes per-block labels as distinct columns and aggregates repeated values, ordered to match journey structure.
- Localization
  - Added translations for “Poll Selection” and “Text Submission.”

### CODE REVIEW

**Review by tanflem — APPROVED** [review]

File: apis/api-journeys-modern/src/schema/journeyVisitor/journeyVisitor.ts [hunk 1]
```diff
@@ -170,11 +174,12 @@ async function* getJourneyVisitors(
...
-        if (event.blockId) {
-          if (row[event.blockId]) {
-            row[event.blockId] += `;${event.value!}`
+        if (event.blockId && event.label) {
+          const key = `${event.blockId}-${event.label}`
+          if (row[key]) {
+            row[key] += `; ${event.value!}`
           } else {
-            row[event.blockId] = event.value!
+            row[key] = event.value!
```
[the remark this answers is not in the document]
Comment (tataihono) [reply apis/api-journeys-modern/src/schema/journeyVisitor/journeyVisitor.ts:182]: this doesn't make sense. the ID is always included in the composite.

**Review by Kneesal — APPROVED** [review]

File: apis/api-journeys-modern/src/schema/journeyVisitor/journeyVisitor.ts [hunk 2]
```diff
@@ -257,11 +269,20 @@ builder.queryField('journeyVisitorExport', (t) => {
           distinct: ['blockId', 'label']
         })
 
+        // TODO: sort blockIds based on order the block appears in the journey
```
[the remark this answers is not in the document]
Comment (tataihono) [reply apis/api-journeys-modern/src/schema/journeyVisitor/journeyVisitor.ts:272]: rip. this comment stopped the functionality from being merged.
